equipment-datastore-common-test/src/main/java/com/telecominfraproject/wlan/equipment/datastore/BaseEquipmentDatastoreTest.java

```diff
@@ -348,6 +348,149 @@ public void testEquipmentPagination()
 
     }
     
+    @Test
+    public void testSearchByMacAndName()
+    {
+       //create 100 Equipments
+       Equipment mdl;
+       int customerId_1 = (int) testSequence.incrementAndGet();
+       
+       int apNameIdx = 0;
+       
+       for(int i = 0; i< 50; i++){
+           mdl = new Equipment();
+           mdl.setCustomerId(customerId_1);
+           mdl.setName("qr_"+apNameIdx);
+           mdl.setProfileId(1);
+           mdl.setLocationId(2);
+           mdl.setEquipmentType(EquipmentType.AP);
+           mdl.setInventoryId("inv-testEquipmentSearchByMacAndName-" + mdl.getName());
+           if (i < 10) {
+        	   mdl.setBaseMacAddress(new MacAddress("A1:FF:FF:FF:FF:0" + i));
+           } else {
+        	   mdl.setBaseMacAddress(new MacAddress("A1:FF:FF:FF:FF:" + i));
```

Comment:
seriously? :)
the condition should be (i<16)
and in both cases :
new MacAddress("A1:FF:FF:FF:FF:" + Integer.toHexString(i));

Reply:
To be fair, the way you have it now works too, but I think it looks weird. 
Given how people are copying pieces of code, in another context this way of mixing base10 and base16 numbers will not work ( for i>99 ).